cuappdev/uplift-backend
# Migrate dev server

## Overview

Added scripts to migrate data in flask

## Changes Made

- Added functionality to manager.py file
- Added app config to app.py to connect to postgres


## Test Coverage

Tested locally using graphql, pgadmin. Testing log here:

1. Delete migrations folder in project to ensure we start from scratch
2. Drop and recreate table in psql: DROP/CREATE DATABASE uplift;
3. Make sure new_column (our dummy column is commented out), Flask run
4. Check in psql that the columns in users do not include our dummy column
5. Add sample data into user object to see if they migrate in graphql playground
6. Check in pgadmin that data has been added
7. Now start migration
8. Kill the running server using ctrl+c
9. Uncomment the dummy column
10. Intialize the migration using: python manager.py db init
11. Migrate using: python manager.py db migrate -m "add_newcol_to_users"
12. Apply the migration using: python manager.py db upgrade
13. Check if migration was successful
14. Flask run to start the server again locally
15. Check in psql the columns in users
16. Check in pgadmin that the data is still persisted
